src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -947,21 +1029,30 @@ codet java_bytecode_convert_methodt::convert_instructions(
     else if(statement=="athrow")
     {
       assert(op.size()==1 && results.size()==1);
-      if(!assertion_failure)
-      {
-        side_effect_expr_throwt throw_expr;
-        throw_expr.add_source_location()=i_it->source_location;
-        throw_expr.copy_to_operands(op[0]);
-        c=code_expressiont(throw_expr);
-        results[0]=op[0];
-      }
-      else
+      code_blockt block;
+      if(!disable_runtime_checks)
       {
-        // TODO: this can be removed once we properly handle throw
-        // if this athrow is generated by an assertion, then skip it
-        c=code_skipt();
-        assertion_failure=false;
+        // TODO throw NullPointerException instead
```

Comment:
This is a TODO for when we will replace all the instrumentation related to runtime exceptions, which currently consists of adding assertions, by throwing exceptions. There are several such TODOs in java_bytecode_convert_method.cpp which we can only address once the exception handling works.